File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> st;
        unordered_set<string> s = {"*","/","-","+"};
        for(string c : tokens){
            if(s.find(c) != s.end()){
                int el2 = st.top();
                st.pop();
                int el1 = st.top();
                st.pop();
                int curr = 0;
                if(c == "*"){
                    curr = el1*el2;
                }
                else if(c == "+"){
                    curr = el1+el2;
                }
                else if(c == "-"){
                    curr = el1-el2;
                }
                else{
                    curr = el1/el2;
                }
                st.push(curr);
            }
            else{
                int num = stoi(c);
                st.push(num);
            }

        }
        return st.top();
    }
};

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for el in tokens:
            if el in ('*',"+","-","/"):
                el2 = stack[-1] 
                stack.pop()
                el1 = stack[-1]
                stack.pop()
                curr = None
                if el == "*":
                    curr = el1*el2
                elif el == "+":
                    curr = el1+el2
                elif el == "-":
                    curr = el1-el2
                else:
                    curr = trunc(el1/el2)
                stack.append(curr)
            else:
                stack.append(int(el))
        return stack[-1]